src/main/java/wooteco/subway/domain/line/Lines.java

```diff
@@ -0,0 +1,50 @@
+package wooteco.subway.domain.line;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jgrapht.graph.WeightedMultigraph;
+
+import wooteco.subway.domain.Section;
+import wooteco.subway.domain.Sections;
+import wooteco.subway.domain.Station;
+import wooteco.subway.domain.path.ShortestPathEdge;
+
+public class Lines {
+    private final List<Line> lines;
+
+    public Lines(List<Line> lines) {
+        this.lines = lines;
+    }
+
+    public Set<Station> extractStations() {
+        Set<Station> stations = new HashSet<>();
+        for (Line line : lines) {
+            Sections sections = line.getSections();
+            addStations(stations, sections);
+        }
+        return stations;
+    }
```

Comment:
```suggestion
    public Set<Station> extractStations() {
        Set<Station> stations = new HashSet<>();
        for (Line line : lines) {
            stations.addAll(line.getStations());
        }
        return stations;
    }
```

line에서 stations를 가져오는 역할을 넘겨주면 어떨까요?
역할을 넘겨준다면 스트림을 활용해서 더욱 깔끔한 코드로도 만들 수 있겠네요 ㅎㅎ